app/controllers/DistributionController.scala

```diff
@@ -62,8 +63,19 @@ class DistributionController @Inject() (
           formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
           value =>
             for {
-              updatedAnswers <- Future.fromTry(request.userAnswers.set(DistributionPage, value))
-              _              <- sessionRepository.set(updatedAnswers)
+              updatedAnswers <-
+                Future.fromTry(
+                  if (value == Distribution.No) {
+                    request.userAnswers
+                      .set(DistributionPage, value)
+                      .flatMap(answer =>
+                        answer.set(DistributionsIncludedPage, DistributionsIncludedForm(DistributionsIncluded.No, None))
+                      )
+                  } else {
+                    request.userAnswers.set(DistributionPage, value)
```

Comment:
Since request.userAnswers.set(DistributionPage, value) is always done you could simplify a bit (either use for comprehension or flatMap, whatever you prefer)

```
for {
   answer <- request.userAnswers
                      .set(DistributionPage, value)
   finalAnswer <- if(value == Distribution.No) answer.set(DistributionsIncludedPage, DistributionsIncludedForm(DistributionsIncluded.No, None)) else answer
} yield finalAnswer
```

Also, if the answer is No, shouldnt we completely delete the DistributionsIncludedPage value?

Reply:
Thats been updated. Setting DistributionsIncludedPage when selecting No so it fixes the issue Raised in Mrc-39 where the Check answers Distribution section doesn't show if No is selected on Distribution first screen